dashboard/app/views/courses/show.html.haml

```diff
@@ -8,6 +8,11 @@
 - data[:hidden_scripts] = @current_user.try(:get_hidden_script_ids, course)
 - data[:show_version_warning] = course.has_older_version_progress?(@current_user) && !course.has_dismissed_version_warning?(@current_user)
 - data[:show_redirect_warning] = redirect_warning
+-# Ask to redirect user to the latest assigned version of this course if they are able to view this course but are not
+-# assigned to it
+- if course.can_view_version?(@current_user) && !@current_user.assigned?(course)
+  - latest_assigned_version = Course.latest_assigned_version(course.family_name, @current_user)
+  - data[:redirect_to_course_url] = latest_assigned_version == course ? nil : latest_assigned_version&.link
```

Comment:
I think there is enough going on in this logic that it should be covered by a unit test. The easiest way to accomplish this is probably to extract it to a helper method in the course model. it could be called via `course.redirect_to_course_url(@current_user)`.

Reply:
good call. i wasn't sure i should put this on the model, but you're definitely right about needing a unit test. adding now!